### ISSUE TITLE
Failed to call calldata_signal_callback for calldata_signal_callback:

### ISSUE THREAD

**Issue by Mazecreator** [vwout/obs-visca-control#2]
I believe I have installed the files in the right place set up the configuration and tried the default port and the configured the Visca port set in the camera. The IP address should be right as well but I am getting these errors back when I switch scenes:

[obs-visca-control.lua] do_cam_action_start (96): Start cam 1 @192.168. action 0 (arg 1)
[obs-visca-control.lua] Failed to call calldata_signal_callback for calldata_signal_callback: ...le/data/obs-plugins/frontend-tools/scripts//ljsocket.lua:417: cannot resolve symbol 'gai_strerror': The specified procedure could not be found.

[obs-visca-control.lua] do_cam_action_start (96): Start cam 1 @192.168. action 2 (arg 1)
[obs-visca-control.lua] Failed to call calldata_signal_callback for calldata_signal_callback: ...le/data/obs-plugins/frontend-tools/scripts//ljsocket.lua:417: cannot resolve symbol 'gai_strerror': The specified procedure could not be found.

[obs-visca-control.lua] do_cam_action_start (96): Start cam 1 @192.168. action 0 (arg 2)
[obs-visca-control.lua] Failed to call calldata_signal_callback for calldata_signal_callback: ...le/data/obs-plugins/frontend-tools/scripts//ljsocket.lua:417: cannot resolve symbol 'gai_strerror': The specified procedure could not be found.

[obs-visca-control.lua] do_cam_action_start (96): Start cam 1 @192.168. action 2 (arg 1)
[obs-visca-control.lua] Failed to call calldata_signal_callback for calldata_signal_callback: ...le/data/obs-plugins/frontend-tools/scripts//ljsocket.lua:417: cannot resolve symbol 'gai_strerror': The specified procedure could not be found.

[obs-visca-control.lua] do_cam_action_start (96): Start cam 1 @192.168. action 0 (arg 2)
[obs-visca-control.lua] Failed to call calldata_signal_callback for calldata_signal_callback: ...le/data/obs-plugins/frontend-tools/scripts//ljsocket.lua:417: cannot resolve symbol 'gai_strerror': The specified procedure could not be found.

[obs-visca-control.lua] do_cam_action_start (96): Start cam 1 @192.168. action 2 (arg 1)
[obs-visca-control.lua] Failed to call calldata_signal_callback for calldata_signal_callback: ...le/data/obs-plugins/frontend-tools/scripts//ljsocket.lua:417: cannot resolve symbol 'gai_strerror': The specified procedure could not be found.

Any idea what to look at to get this working? 

EDIT: Using OBS 27.0.0-rc3 on a Windows 10 PRO PC

**Comment by vwout** [comment]
Hi,

in case of a connection error, there should be a descriptive error message and not a lua failure. This could be related to the lua version shipped with OBS 27 rc3, I will have a look.

But the IP address specified "192.168." is not complete. The last two octets are missing; it should be something like 192.168.x.x.
That probably is the root cause of the failure.

**Comment by Mazecreator** [comment]
Hi @vwout ,

Thanks for taking a look. I will see if I can install a Portable version of V26 to test. I do have the full IP address in the dialog box of 192.168.0.201, I thought that @192.168 was strange but didn't know if it was trying to represent the network (B in this case). I have attached the configuration I am using for your review.  I tried the develop Port and the one set in the camera.  Let me know if you need me to test something!

Regards,
Greg 

![obs64_Y1jjnMlCIh](https://user-images.githubusercontent.com/18412448/120400018-1a67d100-c30b-11eb-8f79-88047e30f0a4.png)

**Comment by vwout** [comment]
Hi @Mazecreator, I tested yesterday with the official release 27.
The issue is not related to RC3, or the version of OBS in general. The lua-error is a bug that I have a solution for (work in progress) but the root cause is the invalid IP address. When you enter the information like in the screenshot above, I expect things to work.

**Comment by Mazecreator** [comment]
Thanks, I can test it when you are ready!

**Comment by vwout** [comment]
You should enter the full and correct IP address; no change is needed for that. I will make improvements regarding the error handling, but that is not a prerequisite for usage by you.

**Comment by Mazecreator** [comment]
> You should enter the full and correct IP address; no change is needed for that. 

I have the 4 octets included, is there another way I need to enter the IP address?  You can see in the screenshot it is 192.168.0.201


Do I need to add:
http://192.168.0.201 or something like that?

**Comment by vwout** [comment]
Just enter `192.168.0.201` as you did in the screenshot. Does it work now?
In case not, can you add the script log?

**Comment by Mazecreator** [comment]
Hummm.... 

Okay, I did a fresh install and entered:
192.168.0.201

Closed the dialog and went to use the tool and failed like posted above.
I went back to the dialog box and it only showed:
192.168.

So I replaced the 0 with a 1 and entered:
192.168.1.201

I then went back and edited the 1 to a 0.  Now it seems to work.  I am wondering if somehow the 0 was parsed as a NULL rather than a number in the octet and truncated the string?

Okay, I think it is working but had to do some dancing to get the proper IP address in there an recognized internally.  Prior to this, I had 192.168.0.201, but I was still getting the error like it was: 192.168.

Thanks for the great plugin.
